### DISCUSSION TITLE
Value is not updated when using setValue and input file field

### DISCUSSION THREAD

**Discussion by lobaak** [react-hook-form/react-hook-form#8612] [Q&A] [answered]
### Version Number

7.33.0

### Codesandbox/Expo snack

https://codesandbox.io/s/muddy-framework-g27puu

### Steps to reproduce

I'm a big fan of the library, thanks for all your hard work!

I'm trying to create drag and drop functionality using a file input's label. Clicking the label and selecting a file will update the value of the input but dragging a file onto the label doesn't do anything. I'm using `setValue` to try and update the value of the input.

I'm probably doing something wrong but couldn't find a working example.

I can see the `onDrop` is being called because the console.log is visible but can't figure out why `setValue` isn't updating the input's value.

```jsx
  const handleDrop = (e) => {
    e.preventDefault();
    e.stopPropagation();
    const { files } = e.dataTransfer;
    if (files?.length > 1) return;

    // This doesn't work
    setValue("file", files);
    console.log(`dropped files`, files, files instanceof FileList);
  };
  ```

### Expected behaviour

When a file is dropped the file input's value is updated

### What browsers are you seeing the problem on?

Chrome

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by bluebill1049** [comment]
Hey, @lobaak my recommendation is to avoid manipulating the file object within the input (include with setValue) and let the input file handle the field object itself considering there is a security restriction on the input file, instead it's easier and makes more sense to build a controlled component and wrapped with `Controller` or `useController`. You can not programmatically update an input file, this is due to security. More info here: https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/file

**Comment by lobaak** [comment, marked as answer]
Hey @bluebill1049. Okay, got it. That actually makes a lot of sense about the security issues. Following your advice I've come up with a solution that works for what we need. I'll add it to the ticket for anyone else that is looking for a similar thing.

Let me know if it's way-off the mark but I think it works fine. The reset is a bit sketchy

https://codesandbox.io/s/muddy-framework-g27puu
